resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -217,14 +217,42 @@ private[spark] class ExecutorPodsLifecycleManager(
     ExecutorExited(exitCode, exitCausedByApp, exitMessage)
   }
 
+  // A utility function to try and help people figure out whats gone wrong faster.
+  private def describeExitCode(code: Int): String = {
+    val humanStr = code match {
+      case 0 => "(success)"
+      case 1 => "(generic, look at logs to clarify)"
+      case 42 => "(douglas adams)"
```

Comment:
Does K8s refer `The Hitchhiker's Guide to the Galaxy` with 42?

Reply:
https://stackoverflow.com/questions/39366400/why-is-it-exit42 :p

Reply:
😆 
Nit: (douglas adams) => (Douglas Adams) especially as Uncaught is capitalised in "(Uncaught exception)".
And the latter is more like villain in our story not a hero :)